pkg/flavorgen/flavor_gen.go

```diff
@@ -0,0 +1,352 @@
+	}
+
+	for _, template := range flavorTemplates {
+		var flavorTemplate hvs.FlavorTemplate
+		flavorJSON, err := ioutil.ReadFile(template)
+		if err != nil {
+			return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Could not load flavor template file")
+		}
+
+		if len(flavorJSON) == 0 {
+			return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Empty flavor template file is given, unable to proceed further")
+		}
+
+		// Validate the format
+		if !json.Valid(flavorJSON) {
+			return hcType.HostManifest{}, nil, errors.New("flavorgen/flavor_gen:processJsonFile() Given flavor template file is not a valid json")
+		}
+
+		err = json.Unmarshal(flavorJSON, &flavorTemplate)
+		if err != nil {
+			return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Could not unmarshal flavor template json")
+		}
+
+		if flavorTemplate.ID == uuid.Nil {
+			flavorTemplate.ID, err = uuid.NewRandom()
+			if err != nil {
+				return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Failed to generate UUID for flavor template")
+			}
+		}
+
+		conditionEval := false
+		for _, condition := range flavorTemplate.Condition {
+			expectedData, err := jsonquery.Query(manifest, condition)
+			if err != nil {
+				return hcType.HostManifest{}, nil, errors.Wrap(err, "flavorgen/flavor_gen:processJsonFile() Failed to query search condition with hostmanifest")
+			}
+			if expectedData == nil {
+				fmt.Println(flavorTemplate.Condition)
+				conditionEval = true
+				break
+			}
+		}
+		if !conditionEval {
+			flavors = append(flavors, flavorTemplate)
+		}
+	}
+	if len(flavors) == 0 {
+		return hcType.HostManifest{}, nil, errors.New("flavorgen/flavor_gen:processJsonFile() Condition does not matches with manifest file")
+	}
+
+	return hostManifest, flavors, nil
+}
+
+// checkIfValidFile to check the given file exists and in proper format
+func checkIfValidFile(filename string) (bool, error) {
+	defaultLog.Trace("flavorgen/flavor_gen:checkIfValidFile() Entering")
+	defer defaultLog.Trace("flavorgen/flavor_gen:checkIfValidFile() Leaving")
+
+	// Checking if entered file is json by using the filepath package
+	if fileExtension := filepath.Ext(filename); fileExtension != ".json" {
+		return false, fmt.Errorf("File %s is not json", filename)
+	}
+
+	// Checking if filepath entered belongs to an existing file.
+	if _, err := os.Stat(filename); err != nil && os.IsNotExist(err) {
+		return false, fmt.Errorf("File %s does not exist", filename)
+	}
+
+	// If we get to this point, it means this is a valid file
+	return true, nil
+}
+
+const helpStr = `Usage:
+
+flavor-gen <command> [arguments]
+	
+Available Commands:
+	-f                     To provide Flavor template json file
+	-m                     To provide Hostmanifest json file
+	-k                     To provide Signing key file

```

Comment:
as flavor API accepts unsigned flavors, this can be removed from flavor gen tool